### PR TITLE
feat(specs): add task input for bigquery

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1104](https://algolia.atlassian.net/browse/DI-1104)

### Changes included:

- add `OnDemandDateUtilsInput` for `onDemand` tasks linked to a `bigquery` source
- add `ScheduleDateUtilsInput` for `schedule` tasks linked to a `bigquery` source

[DI-1104]: https://algolia.atlassian.net/browse/DI-1104?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: specs/ingestion/common/schemas/task.yml [hunk 1]
```diff
@@ -239,3 +247,33 @@ SubscriptionTrigger:
...
+    - startDate
+    - endDate
+
+ScheduleDateUtilsInput:
+  type: object
+  additionalProperties: false
+  description: The input for a `schedule` task whose source is of type `bigquery` and for which extracted data spans a fixed number of days.
+  properties:
+    timeframe:
+      description: The timeframe of the extraction (RFC3339 format).
```
Comment (millotp) [inline specs/ingestion/common/schemas/task.yml:274]:
```suggestion
      description: The timeframe of the extraction, in number of days from today.
```

**Review by millotp — CHANGES_REQUESTED** [review]
Nice !

**Review by millotp — APPROVED** [review]

**Review by shortcuts — APPROVED** [review]

**Comment by Fluf22** [comment]
@millotp @shortcuts  do OpenAPI supports conditional ref?
Otherwise, I thought we could iterate on that to enforce type higher in the chain, with the trigger type discriminating between both task types.

In the meantime, I thought this was enough ☺️

**Comment by shortcuts** [comment]
> @millotp @shortcuts do OpenAPI supports conditional ref? Otherwise, I thought we could iterate on that to enforce type higher in the chain, with the trigger type discriminating between both task types.

That's what we've done in the other specs, e.g. recommend have the whole object of the query as a oneOf, where the enum is the discriminator. It's indeed ideal, but for now since the specs is in early stage it's fine to have looser type and refine later before going out of alpha I believe

**Pull request merged**